homeassistant/components/switch/rachio.py

```diff
@@ -4,227 +4,226 @@
 https://home-assistant.io/components/switch.rachio/
 """
+from abc import abstractmethod
 from datetime import timedelta
 import logging
-
 import voluptuous as vol
 
 from homeassistant.components.switch import PLATFORM_SCHEMA, SwitchDevice
-from homeassistant.const import CONF_ACCESS_TOKEN
 import homeassistant.helpers.config_validation as cv
-import homeassistant.util as util
+from homeassistant.components.rachio import DOMAIN, \
+    EVENT_RACHIO_CONTROLLER_UPDATE, EVENT_RACHIO_ZONE_UPDATE, \
+    KEY_DEVICE_ID, KEY_ENABLED, KEY_ID, KEY_NAME, KEY_ON, KEY_SUBTYPE,\
+    KEY_SUMMARY, KEY_ZONE_ID, KEY_ZONE_NUMBER,\
+    SUBTYPE_ZONE_STARTED, SUBTYPE_ZONE_STOPPED, SUBTYPE_ZONE_COMPLETED,\
+    SUBTYPE_SLEEP_MODE_ON, SUBTYPE_SLEEP_MODE_OFF
 
-REQUIREMENTS = ['rachiopy==0.1.2']
+DEPENDENCIES = ['rachio']
 
 _LOGGER = logging.getLogger(__name__)
 
+# Manual run length
 CONF_MANUAL_RUN_MINS = 'manual_run_mins'
-
-DATA_RACHIO = 'rachio'
-
 DEFAULT_MANUAL_RUN_MINS = 10
 
-MIN_UPDATE_INTERVAL = timedelta(seconds=30)
-MIN_FORCED_UPDATE_INTERVAL = timedelta(seconds=1)
-
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_ACCESS_TOKEN): cv.string,
     vol.Optional(CONF_MANUAL_RUN_MINS, default=DEFAULT_MANUAL_RUN_MINS):
         cv.positive_int,
 })
 
+ATTR_ZONE_SUMMARY = 'Summary'
+ATTR_ZONE_NUMBER = 'Zone number'
+
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Rachio switches."""
-    from rachiopy import Rachio
+    manual_run_time = timedelta(minutes=config.get(CONF_MANUAL_RUN_MINS))
+    _LOGGER.info("Rachio run time is %s", str(manual_run_time))
 
-    # Get options
-    manual_run_mins = config.get(CONF_MANUAL_RUN_MINS)
-    _LOGGER.debug("Rachio run time is %d min", manual_run_mins)
+    # Add all zones from all controllers as switches
+    devices = []
+    for controller in hass.data[DOMAIN].controllers:
+        devices.append(RachioStandbySwitch(hass, controller))
 
-    access_token = config.get(CONF_ACCESS_TOKEN)
+        for zone in controller.list_zones():
+            devices.append(RachioZone(hass, controller, zone, manual_run_time))
 
-    # Configure API
-    _LOGGER.debug("Configuring Rachio API")
-    rachio = Rachio(access_token)
+    add_devices(devices)
+    _LOGGER.info("%d Rachio switch(es) added", len(devices))
 
-    person = None
-    try:
-        person = _get_person(rachio)
-    except KeyError:
-        _LOGGER.error(
-            "Could not reach the Rachio API. Is your access token valid?")
-        return
 
-    # Get and persist devices
-    devices = _list_devices(rachio, manual_run_mins)
-    if not devices:
-        _LOGGER.error(
-            "No Rachio devices found in account %s", person['username'])
-        return
+class RachioSwitch(SwitchDevice):
+    """Represent a Rachio state that can be toggled."""
 
-    hass.data[DATA_RACHIO] = devices[0]
+    def __init__(self, controller, poll=True):
```

Comment:
Remove the polling option and send `True` as second param to add_devices so it will call the `update` method.

Reply:
Same explanation as https://github.com/home-assistant/home-assistant/pull/15111#pullrequestreview-131742582